transport/tchannel/handler.go

```diff
@@ -244,12 +254,17 @@ func (rw *responseWriter) Write(s []byte) (int, error) {
 }
 
 func (rw *responseWriter) Close() error {
-	var err error
+	err := rw.ensureWroteHeaders()
+
 	if rw.bodyWriter != nil {
-		err = rw.bodyWriter.Close()
+		if closeErr := rw.bodyWriter.Close(); err == nil {
```

Comment:
I just realized YARPC is already using `multierr`. We can just use Append here instead.